yapapi/golem.py

```diff
@@ -28,6 +28,37 @@
 
 
 class Golem(_Engine):
+    """
+    The main entrypoint of the Golem's high-level API.
+
+    Provides two methods that reflect the two modes of operation, or two types of jobs
+    that can currently be executed on Golem.
+
+    The first one - `execute_tasks` - instructs Golem to take a sequence of task fragments
+    that the user wishes to compute on Golem and distributes those among the providers.
+
+    The second one - `run_service` - instructs Golem to spawn a certain number of instances
+    of a service based on a single service specification (a specialized implementation
+    inheriting from `yapapi.Service`)
+
+    While the two models are not necessarily completely disjoint - in that we can create a
+    service that exists to process a certain number of computations and, similarly, we can
+    use the task model to run some service - the main difference lies in the lifetime of
+    such job.
+
+    Whereas a task-based job exists for the purpose of computing the specific
+    sequence of tasks and is done once the whole sequence has been processed, the
+    service-based job is created for an indefinite period and the services spawned
+    within it are kept alive as long as they're needed.
+
+    Additionally, the service interface provides a way to easily define handlers for
+    certain, discrete phases of a lifetime of each service instance - startup, running and
+    shutdown.
```

Comment:
Can we link to the `Service` class here? It's docstring should contain more information about the lifecycle of service instances.

Reply:
I'm unsure how to link to another place in the docs, plus, I'd make it a separate PR